src/components/multiscreen/multiscreen.tsx

```diff
@@ -0,0 +1,153 @@
+import {OnClickEvent} from '@playkit-js/common/dist/hoc/a11y-wrapper';
+import {Position} from '../../enums';
+import {PlaykitUI} from 'kaltura-player-js';
+import {MultiscreenPlayer} from '../../types';
+import * as styles from './multiscreen.scss';
+
+const {Tooltip} = KalturaPlayer.ui.components;
+const {withEventManager} = KalturaPlayer.ui.Event;
+const {withText, Text} = KalturaPlayer.ui.preacti18n;
+
+const translates = {
+  multiscreenLabel: <Text id="dualScreen.multiscreen">Show more screens</Text>
+};
+interface MultiscreenProps {
+  players: Array<MultiscreenPlayer>;
+  updateOnStateChanged?: boolean;
+  eventManager?: PlaykitUI.EventManager;
+  multiscreenLabel?: string;
+  getParentRef?: () => RefObject<HTMLDivElement>;
+  getPosition?: () => Position;
+}
+interface MultiscreenState {
+  isOpen: boolean;
+  xPosition: number;
+}
+
+let focusOnMultiscreenButton = false;
+
+@withText(translates)
+@withEventManager
+export class Multiscreen extends Component<MultiscreenProps, MultiscreenState> {
+  private _multiscreenPlayersRefs: Array<RefObject<HTMLDivElement>> = [];
+  private _multiscreenPlayersWrapperRef = createRef<HTMLDivElement>();
+
+  state = {isOpen: false, xPosition: 0};
+
+  componentDidMount(): void {
+    this._attachVideoElements();
+  }
+
+  componentDidUpdate(previousProps: Readonly<MultiscreenProps>, previousState: Readonly<MultiscreenState>): void {
+    if (!previousState.isOpen && this.state.isOpen) {
+      if (this.props.updateOnStateChanged) {
+        // re-attach video elements for SbS layout
+        this._attachVideoElements();
+      }
+      if (this.props.getParentRef && this.props.getPosition) {
+        // check if multiscreen container inside PiP container for left positons
+        if ([Position.TopLeft, Position.BottomLeft].includes(this.props.getPosition())) {
+          const parentLeftPostion = this.props.getParentRef().current?.getBoundingClientRect().left || 0;
+          const multiscreenLeftPostion = this._multiscreenPlayersWrapperRef.current?.getBoundingClientRect().left || 0;
+          const rightShift = multiscreenLeftPostion - parentLeftPostion;
+          if (rightShift < 0) {
+            this.setState({xPosition: rightShift});
+          }
+        } else if (this.state.xPosition !== 0) {
+          this.setState({xPosition: 0});
+        }
+      }
+    }
+  }
+
+  componentWillUnmount() {
+    this._multiscreenPlayersRefs = [];
+  }
+
+  private _attachVideoElements = () => {
+    this._multiscreenPlayersRefs.forEach((ref, index) => {
+      const videoElement = this.props.players[index].player.getVideoElement();
+      videoElement.tabIndex = -1;
+      ref.current!.prepend(videoElement);
+    });
+  };
+
+  private _togglePopover = (event: OnClickEvent, byKeyboard = false) => {
+    // a11y wrapper prevent propagation of click event, so 'click' have to be triggered manually
+    focusOnMultiscreenButton = byKeyboard;
+    document.dispatchEvent(new Event('click'));
```

Comment:
what do you need the click event for ? and why on top level ?

Reply:
https://github.com/kaltura/playkit-js-common/blob/master/src/hoc/a11y-wrapper/index.tsx#L60

Reply:
@SivanA-Kaltura 
[mul](https://user-images.githubusercontent.com/51074448/232538850-2dec79e7-f8d8-4597-8344-53a1dad92c48.png)
Multiscreen has 2 popups, if first one opens - second one have to got closed. Let's assume we already have onpe popup opened, then when user opens second popup, A11y wrapper prevents events propagation, so "onclick" listener doesn't work and first popup continue to be opened. That's why I manually dispatch click to document (the same node has listener for click)

Reply:
I don't really like the idea of firing document-wide events. Is is possible to use blur() ? Does firing the click event work on iframe embeds ?

Reply:
for Iframes it definitely gonna work, because iframe has own scope.